### PR TITLE
Allow access via JWT for fact checking

### PR DESCRIPTION
**Pull request by danielroseman**
This accepts a querystring parameter `token`, consisting of a JWT
containing a subject claim for a fact check id. If the token is valid,
the fact_check_id is passed upstream in the GOVUK_FACT_CHECK_ID header.

Note that this app does not do anything to verify the fact_check_id
itself; that will be done by content-store.

Decoding the token relies on the JWT_AUTH_SECRET env var; the same
value will need to be shared with any publishing app that wants to
generate a token to be used for fact checking.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -14,6 +14,8 @@ An authenticating proxy application that proxies requests to an upstream service
 - **signonotron**: Single signon service for GOV.UK authentication.
 - **`GOVUK_UPSTREAM_URI`**: environment variable used to specify the upstream
   site.
+- **`GOVUK_FACT_CHECK_ID`**: The HTTP header which contains the UID of a fact
```
Comment (kevindew) [inline README.md]: do you mean UUID here?

File: lib/proxy.rb [hunk 2]
```diff
@@ -54,11 +73,15 @@ def authenticate!(env)
...
 
   def add_authenticated_user_header(env)
-    if env['warden']
+    if env['warden'] && env['warden'].user
       env['HTTP_X_GOVUK_AUTHENTICATED_USER'] = env['warden'].user.uid.to_s
     end
   end
 
+  def remove_token_param(env)
+    env['QUERY_STRING'] = URI.encode_www_form(CGI.parse(env['QUERY_STRING']).except("token"))
```
Comment (kevindew) [inline lib/proxy.rb]: minor - I think this could be easier to follow broken into 2 steps

File: README.md (hunk 1 above)
Comment (danielroseman) [inline README.md]: fixed

File: lib/proxy.rb (hunk 2 above)
Comment (danielroseman) [inline lib/proxy.rb]: Hopefully that looks better now.

**Pull request merged**